atc/path_flag.go

```diff
@@ -2,19 +2,40 @@ package atc
 
 import (
 	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
 	"path/filepath"
+	"runtime"
 	"strings"
 
 	"github.com/jessevdk/go-flags"
 )
 
+const tempFilePattern = "___dash-stdin-path-flag___"
+
 type PathFlag string
 
 func (path *PathFlag) UnmarshalFlag(value string) error {
 	if value == "" {
 		return nil
 	}
 
+	if value == "-" && runtime.GOOS != "windows" {
+		tempf, err := ioutil.TempFile("", tempFilePattern)
```

Comment:
I may be wrong, but I interpreted @agurney's [comment](https://github.com/concourse/concourse/pull/5861/files#r454470474) to mean that on Mac and Linux, we can use `/dev/stdin`, but on Windows we'll have to keep making a temp file. So, something like:

```go
if value == "-" {
	if runtime.GOOS != "windows" {
		*path = PathFlag("/dev/stdin")
		return nil
	}
	tempf, err := ioutil.TempFile("", tempFilePattern)
	if err != nil {
		return fmt.Errorf("failed to create a temp file")
	}
	defer tempf.Close()

	_, err = io.Copy(tempf, os.Stdin)
	if err != nil {
		return fmt.Errorf("failed to write temp file: %s", err.Error())
	}
	*path = PathFlag(tempf.Name())
	return nil
}
```

Reply:
Or maybe it would be better to check for the existence of /dev/stdin, rather than checking the GOOS

Reply:
Ah... I misunderstood  @agurney's comment. Just fixed. And checking existence of /dev/stdin also works. I managed to test on a windows VM, and the latest commit worked well.